### PR TITLE
Integration test

### PR DESCRIPTION
**Pull request by Bifabig**
In this pull request we have done the following:

- Fix n+1 problem

   - Before:
  
all_posts_befor

   - After:  
all_posts_after

- As well as integration tests for users#index, users#show, posts#index and posts#show routes.

### CODE REVIEW

File: spec/integration/posts_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,78 @@
+require 'rails_helper'
+
+RSpec.describe 'Posts#alls', type: :system do
+  before(:all) do
+    Comment.delete_all
+    Post.delete_all
+    User.delete_all
+    @tom = User.create(name: 'Tom', photo: 'https://placehold.co/200x133', bio: 'Teacher from Mexico.',
+                       posts_counter: 0)
```
Comment (Olamarx) [inline spec/integration/posts_index_spec.rb]: - [ ] I will encourage you to double-check these tests as none of these tests are passing.
image

File: spec/integration/posts_show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,60 @@
+require 'rails_helper'
+
+RSpec.describe 'Posts#details', type: :system do
+  before(:all) do
+    Comment.delete_all
+    Post.delete_all
+    User.delete_all
+    @tom = User.create(name: 'Tom', photo: 'https://placehold.co/200x133', bio: 'Teacher from Mexico.',
```
Comment (Olamarx) [inline spec/integration/posts_show_spec.rb]: Good job, these tests passed 
image

File: spec/integration/users_index_spec.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    @lilly = User.create(name: 'Lilly', photo: 'https://placehold.co/200x133', bio: 'Teacher from Poland.',
+                         posts_counter: 0)
+    @users = User.all
+  end
+
+  it 'I can see the username of all other users.' do
+    visit '/users'
+    expect(page).to have_content('Tom')
+    expect(page).to have_content('Lilly')
+  end
```
Comment (Olamarx) [inline spec/integration/users_index_spec.rb]: - [ ] This test and the other, as shown in the attached test result are failing, I will encourage to check it again
image

File: spec/integration/users_show_spec.rb [hunk 4]
```diff
@@ -0,0 +1,63 @@
+require 'rails_helper'
+
+RSpec.describe 'Users#shows', type: :system do
+  before(:all) do
+    Comment.delete_all
+    Post.delete_all
+    User.delete_all
+    @tom = User.create(name: 'Tom', photo: 'https://placehold.co/200x133', bio: 'Teacher from Mexico.',
```
Comment (Olamarx) [inline spec/integration/users_show_spec.rb]: - [ ] All these tests too are failing, kindly double-check them.
image

**Review by Olamarx — CHANGES_REQUESTED** [review]
Hi @Bifabig,

Good job so far, indeed you have spent time on this project!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
- Linters passed ✔️
- Readme file professional ✔️
- All tests written ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/integration/posts_index_spec.rb (hunk 1 above)
Comment (rotimiazeez) [reply spec/integration/posts_index_spec.rb]: As requested here, kindly fix all failing tests to pass this milestone 👍🏽

File: spec/integration/users_show_spec.rb (hunk 4 above)
Comment (rotimiazeez) [reply spec/integration/users_show_spec.rb]: Kindly fix these tests too 👍🏽

**Review by rotimiazeez — CHANGES_REQUESTED** [review]
# Hi TEAM 👍🏽,
## Good job so far!
### To Highlight 👇🏽
- N+1 problem solved as requested 👌🏽
- No Linter error ✔️ 
- Good PR description 👍🏽 
- Good commit message(s) 😉
- Descriptive README 📖
You still need to work on some issues to prepare your project for the final evaluation, but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR WORDS>


## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you consider them as they can improve your code._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag (@rotimiazeez) me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid, unless it is requested otherwise._**

File: app/views/shared/_post_card.html.erb [hunk 5]
```diff
@@ -1,7 +1,7 @@
 <div class="card">
   <div class="card-body">
     <h4 class="card-title"><%= post.title %></h4>
-    <div class="text-truncate text-justify"><%= post.text %></div>
+    <div class="text-truncate text-justify"><%= post.excerpt %></div>
```
Comment (DammyShittu) [inline app/views/shared/_post_card.html.erb:4]: - [ ] About 2/3 of your tests are still failing. Kindly evaluate and fix this.

  image

  Here is an example of a failed test. Most of them fail due to this "excerpt" here in the post partial which does not exist, according to the error.

  image

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @Bifabig and @tamoorsaeed22,

## Highlights
✔️ Some passing tests

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/shared/_post_card.html.erb (hunk 5 above)
Comment (Bifabig) [reply app/views/shared/_post_card.html.erb:4]: Thanks @DammyShittu

**Review by Ibizugbe — CHANGES_REQUESTED** [review]
Hi @Team👋🏼👋🏼👋🏼, 

require_changes


Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights✅✅✅

- [x] Some test cases are passing👍🏼.
- [x] Linters are passing👍🏼.

 ## Required Changes ♻️

- [ ] Some of your test cases are still failing, kindly do well to fix them as suggested by the previous reviewers😉. Please ensure all test cases are passing before proceeding to submit, so as to avoid any from of blockers👍🏼.
image


_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by gemmen29 — APPROVED** [review]
Hi @Bifabig, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 



## Optional suggestions

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by KDlamini** [comment]
Hi @Bifabig,

I received a review request for this pull request yet it was already approved and merged about an hour ago. I will invalidate it since this is a double request and unnecessary.

**Comment by Bifabig** [comment]
> Hi @Team👋🏼👋🏼👋🏼,
> 
> require_changes
> 
> Great work on making the changes requested by a previous reviewer 👏 You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.
> 
> ## Highlights✅✅✅
> * [x]  Some test cases are passing👍🏼.
> * [x]  Linters are passing👍🏼.
> 
> ## Required Changes ♻️
> * [ ]  Some of your test cases are still failing, kindly do well to fix them as suggested by the previous reviewers😉. Please ensure all test cases are passing before proceeding to submit, so as to avoid any from of blockers👍🏼.
> 
> image
> _Check the comments under the review._
> 
> ## Optional suggestions
> _Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._
> 
> Cheers and Happy coding!👏👏👏
> 
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
> 
> **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
> 
> _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

@Ibizugbe Our project is approved... but 

> Hi @Bifabig,
> 
> I received a review request for this pull request yet it was already approved and merged about an hour ago. I will invalidate it since this is a double request and unnecessary.

@KDlamini Yes but it's not updated on my dashboard... Why is that?

**Comment by Ibizugbe** [comment]
Hi @Bifabig , sorry for the inconviniences. Kindly reach out to @gemmen29, probably there has been a slight mistake👍🏼.

**Comment by Bifabig** [comment]
> Reference

@gemmen29 You recently approved our pull request but my dashboard has not been updated. Please have a look.

**Comment by Bifabig** [comment]
> Hi @Bifabig , sorry for the inconviniences. Kindly reach out to @gemmen29, probably there has been a slight mistake👍🏼.

@Ibizugbe Is there anything you can do for me... He's not responding right now.

**Comment by tamoorsaeed22** [comment]
@Bifabig Mate it's submitted on my dashboard I think it's just a glitch that is creating the issue and I think if it's not updated on your dashboard we should ask for help from micro verse team

**Comment by enniopo** [comment]
@Bifabig - I marked your project as completed now. 

@tamoorsaeed22 - it´s not a glitch. This project is a pair-programming one but each student is meant to submit their own PR. 
Your request should have been invalidated because you used your partner PR. 
I strongly suggest you to create your own PR for this project.

**Pull request merged**